File: backend/src/main/java/com/ufc/backend/backend/resources/UserResource.java
package com.ufc.backend.backend.resources;

import com.ufc.backend.backend.model.Classes;
import com.ufc.backend.backend.model.Student;
import com.ufc.backend.backend.services.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.servlet.support.ServletUriComponentsBuilder;

import java.util.List;
import java.util.Set;

@RestController
@RequestMapping(value = "/users")
@CrossOrigin
public class UserResource {

    @Autowired
    private UserService service;


    @GetMapping("/{id}")
    public ResponseEntity<Student> findById(@PathVariable String id) {
        return ResponseEntity.ok().body(service.findById(id));
    }

    @GetMapping("/classes")
    public ResponseEntity<Set<Classes>> findClasses() {
        return ResponseEntity.ok().body(service.findAllTakenClasses());
    }
    @GetMapping("/classes/nottaken/")
    public ResponseEntity<Set<Classes>> findClassesNotTaken() {
        return ResponseEntity.ok().body(service.findAllNotTakenClasses());
    }


    @PostMapping
    public ResponseEntity<Student> insert(@RequestBody Student user) {
        return ResponseEntity.created(ServletUriComponentsBuilder
                .fromCurrentRequest()
                .path("/users")
                .buildAndExpand(user.getId())
                .toUri()).body(service.insert(user));
    }

  /*  @DeleteMapping("/{classId}")
    public ResponseEntity<Boolean> deleteClassFromSemester(@PathVariable String classId) {
        service.deleteClassFromSemester(classId, null);
        return ResponseEntity.noContent().build();
    }*/
}

File: backend/src/main/java/com/ufc/backend/backend/resources/UserResource.kt
package com.ufc.backend.backend.resources

import com.ufc.backend.backend.model.subject.Subject
import com.ufc.backend.backend.model.student.Student
import com.ufc.backend.backend.model.student.StudentMapper
import com.ufc.backend.backend.model.student.dto.StudentDto
import com.ufc.backend.backend.model.student.dto.StudentInsertDto
import com.ufc.backend.backend.services.UserService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import org.springframework.web.servlet.support.ServletUriComponentsBuilder

@RestController
@RequestMapping(value = ["/users"])
@CrossOrigin
class UserResource(
    private val service: UserService,
    private val mapper: StudentMapper

) {
    @GetMapping("/{id}")
    fun findById(@PathVariable id: String): ResponseEntity<Student> {
        return ResponseEntity.ok().body(service.findById(id))
    }

    @GetMapping("/subject/{id}")
    fun findClasses(@PathVariable id: String): ResponseEntity<Collection<Subject>> {
        return ResponseEntity.ok().body(service.findAllTakenClasses(id))
    }

    @PostMapping
    fun insert(@RequestBody user: StudentInsertDto): ResponseEntity<StudentDto> {
        return ResponseEntity.created(
            ServletUriComponentsBuilder
                .fromCurrentRequest()
                .path("/users")
                .buildAndExpand(user.email)
                .toUri()
        ).body(mapper.entityToDto(service.insert(user)))
    }
}